### DISCUSSION TITLE
Position an item

### DISCUSSION THREAD

**Discussion by Maikel1963** [RibirX/Ribir#609] [Q&A] [answered]
Hello everybody,
for me, the turorial is jumping a little bit around and I do not grasp every idea. So I am trying to build something from scratch which in the end should contain all ideas I need for my projects. But I am stuck nearly at the beginning.
with this code:
```
//main.rs
mod ui;
use ribir::prelude::*;
use ui::hello_world;

fn main() {
    App::run(hello_world())
    .with_size(Size::new(300., 150.))
    .with_title("Hello");
}

```

```
//ui.rs
use ribir::prelude::*;

pub fn hello_world() -> impl WidgetBuilder {
  fn_widget! {
    @Flex {
      align_items: Align::Stretch,
      justify_content: JustifyContent::Center,
      background: Color::RED,
      @Expanded {     
        @Text {
          text: "Hello World!",        
        }
      }
    }
  }
}

```
I get something that looks like this:
![image](https://github.com/RibirX/Ribir/assets/17568498/a680f2d8-bdcb-4fd3-b621-08a37f2db3e5)

But I want the text (Or whatever there will be in the future) to be positioned in the center of the window.
Does anybody know how to achieve this?

**Comment by M-Adoo** [comment, marked as answer]
You can just try
```
fn_widget!{
  @Text {
         h_align:  HAlign::Center,
         v_align:  VAlign::Center,
         text: "Hello World!",        
   }
}
```

Because `align_items` and `justify_content` are configured to place its children in the center of the Flex, but you don't provide a container for the Flex, so it is extended by its children. `h_align` and `v_align` are used to configure how the widget is placed in its parent, where Flex's parent is the window

**Comment by Maikel1963** [comment]
Thank you for the fast answer,
In the meantime I found it by myself in the blog (Introducing overlay). I wasn't aware that positioning is looking from child to parent. I am used to describe the positioning from parent side.
